lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -0,0 +1,531 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for nbhood.CollapseMaskedNeighbourhoodCoordinate."""
+
+
+import unittest
+
+import iris
+from iris.tests import IrisTest
+from iris.exceptions import CoordinateNotFoundError
+from iris.coords import DimCoord
+
+import numpy as np
+
+from improver.nbhood.use_nbhood import CollapseMaskedNeighbourhoodCoordinate
+from improver.tests.nbhood.nbhood.test_BaseNeighbourhoodProcessing import (
+    set_up_cube)
+from improver.tests.nbhood.use_nbhood. \
+    test_ApplyNeighbourhoodProcessingWithAMask \
+    import set_up_topographic_zone_cube
+
+
+class Test__repr__(IrisTest):
+
+    """Test the __repr__ method of CollapseMaskedNeighbourhoodCoordinate."""
+
+    def test_basic(self):
+        """Test that the __repr__ method returns the expected string."""
+        coord_masked = "topographic_zone"
+        weights = iris.cube.Cube(np.array([1.0]), long_name="weights")
+        result = str(CollapseMaskedNeighbourhoodCoordinate(
+            coord_masked, weights))
+        msg = ("<ApplyNeighbourhoodProcessingWithAMask:"
+               " coord_masked: topographic_zone,"
+               " weights: weights / (unknown)"
+               "                 (-- : 1)>")
+        self.assertEqual(result, msg)
+
+
+class Test_reweight_weights(IrisTest):
+
+    """Test the reweight_weights function."""
+
+    def setUp(self):
+        """Set up a cube."""
+        self.mask = np.array([[[1, 1, 1, 0, 0],
+                               [1, 1, 0, 0, 0],
+                               [1, 0, 0, 0, 0],
+                               [1, 0, 0, 0, 0],
+                               [1, 1, 1, 1, 1]],
+                              [[1, 1, 1, 0, 0],
+                               [1, 1, 0, 0, 0],
+                               [1, 0, 0, 0, 0],
+                               [1, 0, 0, 0, 0],
+                               [1, 1, 1, 1, 1]],
+                              [[1, 1, 1, 0, 0],
+                               [1, 1, 0, 0, 0],
+                               [1, 0, 0, 0, 0],
+                               [1, 0, 0, 0, 0],
+                               [1, 1, 1, 1, 1]]])
+        self.cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_grid_points=5)
+        self.cube = iris.util.squeeze(self.cube)
+        weights_data = np.array([[[0.8, 0.7, 0.0, 0.0, 0.0],
```

Comment:
"self.cube" is not used outside this setup function - should this therefore be a local variable not a class member?

Reply:
Done